src/serac/physics/solid_mechanics.hpp

```diff
@@ -828,31 +877,15 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
   template <int... active_parameters, typename MaterialType, typename StateType = Empty>
   void setMaterial(DependsOn<active_parameters...>, MaterialType material, qdata_type<StateType> qdata = EmptyQData)
   {
+    MaterialStressFunctor<MaterialType> material_functor(material, geom_nonlin_);
     residual_->AddDomainIntegral(
         Dimension<dim>{},
         DependsOn<0, 1,
                   active_parameters + NUM_STATE_VARS...>{},  // the magic number "+ NUM_STATE_VARS" accounts for the
                                                              // fact that the displacement, acceleration, and shape
                                                              // fields are always-on and come first, so the `n`th
                                                              // parameter will actually be argument `n + NUM_STATE_VARS`
-        [material, geom_nonlin = geom_nonlin_](double /*t*/, auto /*x*/, auto& state, auto displacement,
-                                               auto acceleration, auto... params) {
-          auto du_dX   = get<DERIVATIVE>(displacement);
-          auto d2u_dt2 = get<VALUE>(acceleration);
-
-          auto stress = material(state, du_dX, params...);
-
-          auto dx_dX = 0.0 * du_dX + I;
-
-          if (geom_nonlin == GeometricNonlinearities::On) {
-            dx_dX += du_dX;
-          }
-
-          auto flux = dot(stress, transpose(inv(dx_dX))) * det(dx_dX);
-
-          return serac::tuple{material.density * d2u_dt2, flux};
-        },
-        mesh_, qdata);
+        std::move(material_functor), mesh_, qdata);
```

Comment:
I'm confused at this call to std::move - material functor has no move constructor or assignment operator.